pkg/deploy/controller/image_change_controller.go

```diff
@@ -80,6 +82,31 @@ func (c *ImageChangeController) HandleImageRepo() {
 	}
 }
 
+// triggerMatchesImages decides whether a given trigger for config matches the provided image repo.
+// When matching:
+// * The trigger From field is preferred over the deprecated RepositoryName field.
+// * The namespace of the trigger is preferred, over the config namespace.
+// * The repo Status.DockerImageRepository field is preferred over the repo DockerImageRepository field.
+func triggerMatchesImage(config *deployapi.DeploymentConfig, trigger *deployapi.DeploymentTriggerImageChangeParams, repo *imageapi.ImageRepository) bool {
+	if len(trigger.From.Name) > 0 {
+		namespace := trigger.From.Namespace
+		if len(namespace) == 0 {
+			namespace = config.Namespace
+		}
+
+		return repo.Namespace == namespace && repo.Name == trigger.From.Name
+	}
+
+	if len(trigger.RepositoryName) > 0 {
```

Comment:
Invert this - do the simple check "empty" and return false, then do the more complex check in the final block.